addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/GitSteps.java

```diff
@@ -1,25 +1,91 @@
+import io.fabric8.launcher.osio.EnvironmentVariables;
 import io.fabric8.launcher.osio.projectiles.OsioProjectile;
+import io.fabric8.launcher.service.git.api.DuplicateHookException;
 import io.fabric8.launcher.service.git.api.GitRepository;
 import io.fabric8.launcher.service.git.api.GitService;
 
+import static io.fabric8.launcher.core.api.events.StatusEventType.GITHUB_CREATE;
+import static io.fabric8.launcher.core.api.events.StatusEventType.GITHUB_PUSHED;
+import static io.fabric8.launcher.core.api.events.StatusEventType.GITHUB_WEBHOOK;
+import static java.util.Collections.singletonMap;
+
 /**
  * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
  */
 @RequestScoped
 public class GitSteps {
 
+    private static final Logger log = Logger.getLogger(GitSteps.class.getName());
+
     @Inject
     private GitService gitService;
 
+    @Inject
+    private Event<StatusMessageEvent> statusEvent;
 
     public GitRepository createRepository(OsioProjectile projectile) {
-        return null;
+        GitRepository gitRepository;
+        final String repositoryName = Objects.toString(projectile.getGitRepositoryName(), projectile.getOpenShiftProjectName());
+        final String repositoryDescription = projectile.getGitRepositoryDescription();
+        gitRepository = gitService.createRepository(repositoryName, repositoryDescription);
+        statusEvent.fire(new StatusMessageEvent(projectile.getId(), GITHUB_CREATE,
+                                                singletonMap("location", gitRepository.getHomepage())));
+        return gitRepository;
     }
 
+    public void pushToGitRepository(OsioProjectile projectile, GitRepository repository) {
+        if (projectile.getStartOfStep() <= StatusEventType.GITHUB_PUSHED.ordinal()) {
+            Path projectLocation = projectile.getProjectLocation();
 
-}
+            // Add logged user in README.adoc
+            Path readmeAdocPath = projectLocation.resolve("README.adoc");
+            if (Files.exists(readmeAdocPath)) {
+                try {
+                    String content = new String(Files.readAllBytes(readmeAdocPath));
+                    String newContent = content.replace("${loggedUser}", gitService.getLoggedUser().getLogin());
+                    Files.write(readmeAdocPath, newContent.getBytes());
+                } catch (IOException e) {
+                    log.log(Level.SEVERE, "Error while replacing README.adoc variables", e);
+                }
+            }
+
+            gitService.push(repository, projectLocation);
+        }
+        statusEvent.fire(new StatusMessageEvent(projectile.getId(), GITHUB_PUSHED));
+    }
+
+    /**
+     * Creates a webhook on the github repo to fire a build / deploy when changes happen on the project.
+     */
+    public void createWebHooks(OsioProjectile projectile, GitRepository gitRepository) {
+        String jenkinsWebhookURL = EnvironmentVariables.ExternalServices.getJenkinsWebhookURL();
```

Comment:
Don't think you can do this as there are multiple jenkins servers, depending on the user this url can be different

Reply:
Nope, this changed in the last couple of months 😁. Now Jenkins is a fixed URL. See https://github.com/fabric8-launcher/launcher-backend/commit/c897fae072566c729ff25cc25adde561b0690e2b